### DISCUSSION TITLE
The id_ed25519-github.pub file is empty after running ssh-keygen

### DISCUSSION THREAD

**Discussion by shawn-hsieh-sndk** [community/community#60005] [Programming Help]
### Select Topic Area

Question

### Body

I have a strange problem and would like to ask for help when I generating the ED25519 ssh key. The id_ed25519-github.pub file is empty after running ssh-keygen -t ed25519 -q -N "" -f ~/.ssh/id_ed25519-github.
Thanks very much.

***:~/.ssh> ll

-rw-------  1 *****                    0 Jul  4 08:23 id_ed25519-github
-rw-r--r--  1 *****                    0 Jul  4 08:23 id_ed25519-github.pub

***:~> ssh -V
OpenSSH_7.4p1, OpenSSL 1.0.2k-fips  26 Jan 2017

**Comment by nirav-gajera** [comment]
1. Check Permissions: Ensure that you have proper permissions to create files in the ~/.ssh directory. Make sure you are the owner of the ~/.ssh directory and have write permissions.  `chmod 700 ~/.ssh
`
2. Delete Existing Key Files: Before generating a new key,  
 `rm ~/.ssh/id_ed25519-github
rm ~/.ssh/id_ed25519-github.pub
`
3.  Generate the ED25519 Key Again: Run the ssh-keygen command again to generate the ED25519 key pair
 `ssh-keygen -t ed25519 -q -N "" -f ~/.ssh/id_ed25519-github
`
4. Verify the Key Generation: After running the ssh-keygen command, check if the id_ed25519-github and id_ed25519-github.pub files are not empty and have proper permissions 
 `ls -l ~/.ssh/id_ed25519-github*
`

**Reply by shawn-hsieh-sndk** [reply]
Hi Nirav,
Thanks for the support.
I have the permission to send the command "chmod 7.. ~/.ssh".
I have followed the step you suggested as below, but the files are still empty.

**:~> chmod 700 /.ssh
***:> rm ~/.ssh/id_ed25519
**:~> ll /.ssh
***:> ssh-keygen -t ed25519 -q -N "" -f /.ssh/id_ed25519-github
***:> ls -l ~/.ssh/id_ed25519-github
-rw------- 1 *** 0 Jul 4 11:16 /home/shawn.hsieh/.ssh/id_ed25519-github
-rw-r--r-- 1 *** 0 Jul 4 11:16 /home/shawn.hsieh/.ssh/id_ed25519-github.pub

Regards,
Shawn

**Reply by shawn-hsieh-sndk** [reply]
I don't know why some tests get messed up after sending, so I'll resend them with picture.
![image](https://github.com/community/community/assets/104541715/f6fc5a21-3f75-46f8-8f38-cfa3ce5d3758)

**Reply by nirav-gajera** [reply]
ssh-keygen -t ed25519 -q -N "" -f ~/.ssh/id_ed25519-github
chmod 700 ~/.ssh
chmod 600 ~/.ssh/id_ed25519-github
cat ~/.ssh/id_ed25519-github

**Reply by shawn-hsieh-sndk** [reply]
The situation is still the same, the files are empty.
![image](https://github.com/community/community/assets/104541715/392873ec-85dd-4ab5-8e29-8f3efa6d9a98)

**Comment by samus-aran** [comment]
Welcome to the GitHub Community, @shawn-hsieh-wdc , we're happy you're here! You are more likely to get a useful response if you are posting your question(s) in the applicable category and are explicit about what your project entails--giving a few more details might help someone give you a nudge in the right direction. I've gone ahead and moved it for you. Good luck!
